pkg/endpoint/endpoint.go

```diff
@@ -104,10 +104,19 @@ func Listen(ctx context.Context, config Config) (ETCDConfig, error) {
 	endpoint := endpointURL(config, listener)
 	logrus.Infof("Kine available at %s", endpoint)
 
+	// Return a empty tls config if we do not receive any tls config
+	if config.ServerTLSConfig == (Config{}.ServerTLSConfig) {
+		return ETCDConfig{
+			LeaderElect: leaderelect,
+			Endpoints:   []string{endpoint},
+			TLSConfig:   tls.Config{},
+		}, nil
+	}
+
 	return ETCDConfig{
 		LeaderElect: leaderelect,
 		Endpoints:   []string{endpoint},
-		TLSConfig:   tls.Config{},
+		TLSConfig:   config.ServerTLSConfig,
```

Comment:
I'm not sure this is correct. The TLSConfig in this block is intended to be used by clients - it should contain the CA chain and **client** cert+key that clients will use when connecting to the server. The ServerTLSConfig contains the **server** cert+key that is used for the listener. 

The best thing to do would probably be to return a new `tls.Config` with the CA cert from the ServerTLSConfig in the trusted root cert list. The client does not need a valid client cert+key, as kine does not currently support client certificate auth (see https://github.com/k3s-io/kine/issues/272#issuecomment-1917767824) - that is probably something we should fix at some point.